ui/src/views/Provisioners/ViewWorkers/index.jsx

```diff
@@ -59,6 +60,23 @@ export default class ViewWorkers extends Component {
     selectedAction: null,
   };
 
+  componentDidMount() {
+    const query = parse(this.props.location.search);
+
+    Object.keys(query).length === 0 ? null : this.deepLinkFilterDropdown();
+  }
+
+  deepLinkFilterDropdown() {
+    const {
+      data: { refetch },
+    } = this.props;
+
+    refetch({
+      quarantined: true,
+    });
```

Comment:
This would cause two queries to run (the initial query and then this refetch) when a query is specified. When the user first loads the page, I think we should be able to run only one query instead of two.

Reply:
Oh yes. I will work on it.